package/MDAnalysis/core/topologyattrs.py

```diff
@@ -31,6 +31,8 @@
 from .topologyobjects import TopologyGroup
 from . import selection
 from . import flags
+from .groups import (GroupBase, Atom, Residue, Segment,
```

Comment:
Yea, good point about the circular imports. I thought about that too.
I like the `Level` approach, but found it insufficient because it doesn't distinguish between singular/plural. In this particular case, we can't use it to specify the transplants/target_levels.

Of course this is all mostly cosmetic, but if you see a way of getting a `Level` with proper order information (and that can be hashed as such) I'd vote for that way. Otherwise I can just revert to using strings to hash.